coordinator/tributary/src/blockchain.rs

```diff
@@ -217,7 +227,11 @@ impl<D: Db, T: TransactionTrait> Blockchain<D, T> {
     commit: Vec<u8>,
     schema: N::SignatureScheme,
   ) -> Result<(), BlockError> {
-    self.verify_block::<N>(block, schema)?;
+    // verify the block, ignore the NonLocalProvided error.
+    let res = self.verify_block::<N>(block, schema);
+    if res.as_ref().is_err_and(|e| !matches!(e, BlockError::NonLocalProvided(_))) {
```

Comment:
This is unsafe as a `NonLocalProvided` error may be raised midway through. If it is, the block would pass as valid here, despite half of the checks not actually executing.

We need to pass in a flag to not perform these checks specifically, yet still perform all other checks, and check that doesn't have any error.